src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -242,6 +252,21 @@ private FormData prepareFormDataSet(FormFields form) {
 				currentList = submitFields;
 			}
 			for (String value : getValues(field)) {
+					//Proceed if the field has a value attribute present(Predefined value)
+					if(field.getFormControl().getAttributesMap().containsKey("value") && 
+							!field.getFormControl().getAttributesMap().get("value").equalsIgnoreCase("submit")	)
```

Comment:
Missed this change while reviewing yesterday (outdated content), it's fine to let the submit fields pass to the value generators.